src/utils/formikHelpers.ts

```diff
@@ -0,0 +1,34 @@
+import { NotificationsService } from "@hooks";
+import { FormikHelpers } from "formik";
+import networkErrorToStringArray from "./networkErrorToStringArray";
+
+export async function execSubmitWithToast<T>(
+  values: T,
+  helpers: FormikHelpers<T>,
+  submitFunc: (values: T, helpers: FormikHelpers<T>) => Promise<void>,
+  notificationsService: NotificationsService,
+  showErrorModalFunc: (value: string[]) => void,
+  entityName: string,
+): Promise<void> {
+  notificationsService.promise(submitFunc(values, helpers), {
+    pending: `Saving a ${entityName}...`,
+    success: `${entityName} saved successfully`,
+    error: (error) => {
+      const errMessage = `Unable to save ${entityName}. An error occurred.`;
```

Comment:
@Exmirai here please follow the same logic like in the case with Saving... message